zebra-consensus/src/error.rs

```diff
@@ -47,6 +47,9 @@ pub enum TransactionError {
     #[error("transaction version number MUST be >= 4")]
     WrongVersion,
 
+    #[error("transaction version {0} not supported by the network upgrade {1:?}")]
+    UnsupportedByNetworkUpgrade(usize, zebra_chain::parameters::NetworkUpgrade),
```

Comment:
Spec: the transaction version is currently 31 bits long, so let's use `u32` here:
```suggestion
    UnsupportedByNetworkUpgrade(u32, zebra_chain::parameters::NetworkUpgrade),
```
https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus